packages/zoe/README.md

```diff
@@ -124,3 +132,25 @@ harden(prepare);
 ```
 
 For an example contract upgrade, see the test at https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js .
+
+### Durability
+
+The contract must retain in durable storage anything that must persist between incarnations. All other state will be lost.
+
+### Kinds
+
+The contract defines the kinds that are held in durable storage. Thus the function calls that define the kinds must be run before the objects are deserialized from durable storage.
+
+# Crank
```

Comment:
Maybe describe this section as `prepare Must Finish Promptly`? I don't know how much we explain "cranks" in our docs, and whether we can rely upon the reader knowing what this jargon means.

Rather than saying "prepare must return in one crank", I might say "while `prepare` is allowed to return a Promise, this Promise must resolve promptly, without awaiting any messages sent to other contracts, vats, or zoe. Otherwise, contract upgrade will fail with a message about `buildRootObject unresolved`."

Do contracts need to contact other vats during the `prepare` of their initial incarnation? If not, the above rule feels pretty easy to explain. But if they do, then we have to explain that the first incarnation is special and more capable, and that sounds tricky to me: I expect users to write their v2 `prepare` by editing their original v1 version, and if they have to also change that version to stop making external calls, I'd expect them to get it wrong a lot.

Reply:
> I don't know how much we explain "cranks" in our docs

We don't and I think we need to. While developing contracts I've found it necessary to consider crank boundaries. And now with upgrade it's especially salient.

> and whether we can rely upon the reader knowing what this jargon means.

Not yet but I hope this section leaves them with some understanding. If we want a more accessible term for the heading/requirement, "Promptness" is a good candidate but can be interpreted in many ways. WDYT of "Locality" in that all the calls have to be within the vat?